### PR TITLE
Make Tensorflow Serving APIs available out of tree

### LINKED ISSUE #267: Make tensorflow_serving.apis and friends available out of tensorflow serving tree

### PR DESCRIPTION
**Pull request by lissyx**
Fixes #267

### CODE REVIEW

File: local_tf/codegen.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import os
+import shutil
+import tempfile
+
+tensorflow_root         = sys.argv[1]
+tensorflow_serving_root = sys.argv[2]
+
+if not os.path.isdir(tensorflow_root) or not os.path.isdir(tensorflow_serving_root):
+    print "No tensorflow_root (%s) or no tensorflow_serving_root (%s)" % (tensorflow_root, tensorflow_serving_root)
+    sys.exit(1)
```
Comment (tilmankamp) [inline local_tf/codegen.py]: Please add a bit of guidance on where those directories typically reside or what to do to get them.

File: local_tf/codegen.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+def myvisit(a, dir, files):
+    if os.path.abspath(dir) == os.path.abspath(tmpdir):
+        return
+
+    initpy = os.path.join(dir, '__init__.py')
+    try:
+        os.utime(initpy, None)
+    except Exception:
+        open(initpy, 'wb').close()
```
Comment (lissyx) [inline local_tf/codegen.py]: well, not my implementation, I found it on some sample in pydoc I think :)

File: local_tf/codegen.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+from grpc.tools import protoc
+
+import sys
+import os
+import shutil
+import tempfile
+
+tensorflow_root         = sys.argv[1]
+tensorflow_serving_root = sys.argv[2]
```
Comment (lissyx) [inline local_tf/codegen.py]: Those needs to be the git repos on a developer's system ; not the installation path. The codegen.py is supposed to be run rarely, to be in sync with the Tensorflow Serving APIs requirements, and then checked-in.

Comment (tilmankamp) [inline local_tf/codegen.py]: OK - but please mention this on the help line below.

Comment (lissyx) [inline local_tf/codegen.py]: done

**Comment by tilmankamp** [comment]
Perfect

**Pull request merged**